Microsoft.Toolkit.Uwp.UI.Controls/MetadataControl/MetadataControl.cs

```diff
@@ -0,0 +1,194 @@
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Documents;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Display <see cref="MetadataUnit"/>s separated by bullets.
+    /// </summary>
+    [TemplatePart(Name = TextContainerPart, Type = typeof(TextBlock))]
+    public sealed class MetadataControl : Control
+    {
+        /// <summary>
+        /// The DP to store the <see cref="Separator"/> property value.
+        /// </summary>
+        public static readonly DependencyProperty SeparatorProperty = DependencyProperty.Register(
+            nameof(Separator),
+            typeof(string),
+            typeof(MetadataControl),
+            new PropertyMetadata(" • ", OnPropertyChanged));
+
+        /// <summary>
+        /// The DP to store the <see cref="AccessibleSeparator"/> property value.
+        /// </summary>
+        public static readonly DependencyProperty AccessibleSeparatorProperty = DependencyProperty.Register(
+            nameof(AccessibleSeparator),
+            typeof(string),
+            typeof(MetadataControl),
+            new PropertyMetadata(", ", OnPropertyChanged));
+
+        /// <summary>
+        /// The DP to store the <see cref="MetadataUnits"/> property value.
+        /// </summary>
+        public static readonly DependencyProperty MetadataUnitsProperty = DependencyProperty.Register(
+            nameof(MetadataUnits),
+            typeof(IEnumerable<MetadataUnit>),
+            typeof(MetadataControl),
+            new PropertyMetadata(null, OnMetadataUnitsChanged));
+
+        private const string TextContainerPart = "TextContainer";
+
+        private TextBlock _textContainer;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MetadataControl"/> class.
+        /// </summary>
+        public MetadataControl()
+        {
+            DefaultStyleKey = typeof(MetadataControl);
+            ActualThemeChanged += OnActualThemeChanged;
+        }
+
+        /// <summary>
+        /// Gets or sets the separator to display between the <see cref="MetadataUnit"/>.
+        /// </summary>
+        public string Separator
+        {
+            get => (string)GetValue(SeparatorProperty);
+            set => SetValue(SeparatorProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the separator that will be used to generate the accessible string representing the control content.
+        /// </summary>
+        public string AccessibleSeparator
+        {
+            get => (string)GetValue(AccessibleSeparatorProperty);
+            set => SetValue(AccessibleSeparatorProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets he <see cref="MetadataUnit"/> to display in the control.
+        /// If it implements <see cref="INotifyCollectionChanged"/>, the control will automatically update itself.
+        /// </summary>
+        public IEnumerable<MetadataUnit> MetadataUnits
```

Comment:
Should this just be `Items` and `MetadataItem` as the type?

Part of me wants this to be an ItemsControl, but I know the templating is probably overkill as we're just trying to use TextBlock... if only it wasn't sealed.

I'm also wondering if this is just an attachedproperty/extension to a TextBlock that just accepts the collection of items, kind of what we're doing with the `SurfaceDialOptions` in #3743. FYI @Sergio0694.

Do we use a different name like `LinkList` and it takes a collection `LinkListInfo` (or Data or Nodes, we should maybe reach out to someone from WinUI about their naming conventions for things that aren't "Item"s)?

```xml
<TextBlock>
  <ui:TextBlockExtensions.LinkList>
     <ui:LinkList Items="{Binding LinkListInfoItems}"
                         Separator="-"
                         HyperlinkStyle="{StaticResource MyHyperlinkStyle}"/>
   </ui:TextBlockExtensions.LinkList>
</TextBlock>
```

Reply:
@vgromfeld think this got side tracked when we went to ship 7.0, thoughts on this approach?

Reply:
Oh sorry, I missed the previous ping 😅
So, yeah I don't see why the same thing couldn't also be done through an attached property, as in, that would technically be doable. I think there would be a few things to consider in that case though (other than the different XAML syntax):
1) The options in `SurfaceDialOptions`, as of today, do not support bindings. As in, they're just meant to be statically initialized once, and then they don't monitor for changes or anything. In this case, we should consider whether developers would instead want to be able to at least change the referenced link items over time or not.
2) Possibly a minor point but worth noting - the attached property would conflict with both other attached properties on the same `TextBlock` changing contents, as well with the actual `Text` contents, if set. Not a big deal, but we would need to make sure to call out this behavior so that devs won't ever attempt to combine this with eg. other custom content, as that would likely cause issues or at least potentially unexpected behavior depending on the internal implementation.

Reply:
Yeah, ideally in WinUI 3 world we could just inherit from TextBlock in the future... 🙂

Reply:
I've renamed the properties to  `Items `and `MetadataItem` and added a new `TextBlockStyle` property as a pass-thru.

I sticked to a control and not an attached property to prevent the `TextBlock.Text` property from being altered by mistake.